python/GafferImageUI/ImageViewUI.py

```diff
@@ -275,80 +298,193 @@ def _hsvString( color ) :
-		viewportGadget.mouseMoveSignal().connect( Gaffer.WeakMethod( self.__mouseMove ), scoped = False )
+			self.__rgbLabel = GafferUI.Label()
+			self.__rgbLabel._qtWidget().setFixedWidth( labelFontMetrics.width( "RGBA : 99999 99999 99999 99999" ) )
 
-		imageGadget = viewportGadget.getPrimaryChild()
-		imageGadget.buttonPressSignal().connect( Gaffer.WeakMethod( self.__buttonPress ), scoped = False )
-		imageGadget.dragBeginSignal().connect( Gaffer.WeakMethod( self.__dragBegin ), scoped = False )
-		imageGadget.dragEndSignal().connect( Gaffer.WeakMethod( self.__dragEnd ), scoped = False )
+			self.__hsvLabel = GafferUI.Label()
+			self.__hsvLabel._qtWidget().setFixedWidth( labelFontMetrics.width( "HSV : 99999 99999 99999" ) )
+
+			self.__exposureLabel = GafferUI.Label()
+			self.__exposureLabel._qtWidget().setFixedWidth( labelFontMetrics.width( "EV : 19.9" ) )
+
+			if mode == 2:
+				m = IECore.MenuDefinition()
+				m.append( "/Pixel Inspector",
+					{ "command" : functools.partial( Gaffer.WeakMethod( self.__addClick ), False ) }
+				)
+				m.append( "/Region Inspector",
+					{ "command" : functools.partial( Gaffer.WeakMethod( self.__addClick ), True ) }
+				)
+				button = GafferUI.MenuButton( "", "plus.png", hasFrame=False, menu = GafferUI.Menu( m, title = "Add Color Inspector" ) )
+			else:
+				button = GafferUI.Button( "", "delete.png", hasFrame=False )
+				button.clickedSignal().connect( Gaffer.WeakMethod( self.__deleteClick ), scoped = False )
+
+
+		self.__pixel = imath.V2i( 0 )
+
+
+		if plug.getName() == "ColorInspectorPlug":
+			viewportGadget = plug.node().viewportGadget()
+			viewportGadget.mouseMoveSignal().connect( Gaffer.WeakMethod( self.__mouseMove ), scoped = False )
+
+			imageGadget = viewportGadget.getPrimaryChild()
+			imageGadget.buttonPressSignal().connect( Gaffer.WeakMethod( self.__buttonPress ), scoped = False )
+			imageGadget.dragBeginSignal().connect( Gaffer.WeakMethod( self.__dragBegin ), scoped = False )
+			imageGadget.dragEnterSignal().connect( Gaffer.WeakMethod( self.__dragEnter ), scoped = False )
+			imageGadget.dragMoveSignal().connect( Gaffer.WeakMethod( self.__dragMove ), scoped = False )
+			imageGadget.dragEndSignal().connect( Gaffer.WeakMethod( self.__dragEnd ), scoped = False )
+
+		self.__swatch.buttonPressSignal().connect( Gaffer.WeakMethod( self.__buttonPress ), scoped = False )
+		self.__swatch.dragBeginSignal().connect( Gaffer.WeakMethod( self.__dragBegin ), scoped = False )
+		self.__swatch.dragEndSignal().connect( Gaffer.WeakMethod( self.__dragEnd ), scoped = False )
+
+		plug.node()["colorInspector"]["evaluator"]["pixelColor"].getInput().node().plugDirtiedSignal().connect( Gaffer.WeakMethod( self._updateFromImageNode ), scoped = False )
+
+		plug.node().plugDirtiedSignal().connect( Gaffer.WeakMethod( self._plugDirtied ), scoped = False )
+		plug.node()["in"].getInput().node().scriptNode().context().changedSignal().connect( Gaffer.WeakMethod( self._updateFromContext ), scoped = False )
```

Comment:
Seems a pity that this is necessary, and I can't figure out why it is even after https://github.com/GafferHQ/gaffer/commit/78aa7b7ded1bd485a73cb9680bb2e06556e4ee67. Probably one for me to look at rather than bother you over though.

Reply:
Yeah, I definitely don't feel like diving into this one right now.